content/articles/building-a-custom-stopwatch-npm-library-in-typescript/index.md

```diff
@@ -0,0 +1,330 @@
+npx tsc
+```
+
+> Note: to have access to the Typescript compiler (tsc), you must have installed Typescript globally in your local machine or as a dependency in the project. If you haven’t, go ahead and run the command below to install Typescript globally:
+
+```bash
+npm install -g typescript
+```
+
+The command will compile The Typescript files to JavaScript thereby producing two new files (`index.js` and `stopwatch.js`) in the `dist` folder. let us go ahead and test/consume the stopwatch package locally.
+To do that, we create a `test` folder in the package root directory (`my-custom-stopwatch`). In the `test` folder, create two files (`index.html` and `test.js`).
+Once that is completed, open the `test.js` file and then import and use the exported stopwatch functions as shown below:
+
+```JavaScript
+import {
+  startWatch,
+  watchcycle,
+  stopWatch,
+  resetWatch,
+} from "../dist/stopwatch";
+
+const timer = document.getElementById("my_stopwatch");
+
+let hours = 0;
+let minutes = 0;
+let seconds = 0;
+let milliseconds = 0;
+let timerStop = true;
+watchcycle(timerStop, hours, minutes, seconds, milliseconds, timer);
+startWatch(timerStop, watchcycle);
+stopWatch(timerStop);
+resetWatch(hours, minutes, seconds, milliseconds, timer);
+
+ ```
+From the snippet above, we:
+  - imported the `startWatch`, `stopWatch`, `watchCycle`, and `resetWatch` from the newly compiled JavaScript files in the `dist` folder.
+  - We then initialized the values of `hours`, `minutes`, `seconds`, and `milliseconds` to zero (0). 
+  - Finally, we passed the initialized values to our imported function which is used to implement the stopwatch logic.
+
+Let us go ahead and setup the `index.html` file to render the stopwatch. To do that, implement the snippet in the `index.html` file as shown below:
+
+```HTML
+<html>
+  <head>
+    <title>My amazing Stopwatch</title>
+  </head>
+  <body>
+    <div id="my_stopwatch">00:00:00</div>
+
+    <ul id="watch_buttons">
+      <li><button onclick="stopWatch()">Stop</button></li>
+      <li><button onclick="startWatch()">Start</button></li>
+
+      <li><button onclick="resetWatch()">Reset</button></li>
+    </ul>
+
+    <script type="module" src="test.js"></script>
+  </body>
+</html>
+```
+We have created a basic stopwatch structure that includes 3 buttons with an `onclick` handler to execute the start, stop and reset operations. We also created a `div` tag with an `id="my_stopwatch"` as required in the `test.js` file.
+Now open the `index.html` file in the browser to view and interact with the amazing stopwatch.
+To do that, open your command terminal and run the command below:
+
+```bash
+cd test
+start index.html
+```
+The command above will open the stopwatch in your default browser.
+If that did not work as expected, install the `http-server` package by running the command below:
+
+```bash
+npm install http-server -g
+```
+Once that is completed, start the server by running:
+
+```bash
+cd test
+http-server
+```

```

Comment:
Hello @davekola 

You didn't test out your code. There are typos in your code. The serve command doesn't work. This is because you can't access the `../dist` directory when you serve your code inside the test folder.

To avoid this, please copy the `stopwatch.js` file inside your test directory and reference it in your `test.js`as follows:

```js
import { ... } from "./stopwatch.js";
```

Reply:
@geoffrey45  done as requested